sycl/test/usm/clext.cpp

```diff
@@ -14,6 +14,37 @@
 
 using namespace cl::sycl::detail::usm;
 
+static int SIZE = 6;
+
+bool findPlatformAndDevice(cl_device_type deviceType,
+			   cl_platform_id& platformOut,
+			   cl_device_id& deviceOut) {
+  cl_platform_id platforms[SIZE];
+  cl_uint numPlatforms;
+  cl_device_id device;
+  cl_int errorCode;
+  bool foundDevice = false;
+
+  errorCode = clGetPlatformIDs(SIZE, platforms, &numPlatforms);
```

Comment:
Here is the idiom to get the list of available platforms:
```c++
cl_uint numPlatforms;
cl_int errorCode = clGetPlatformIDs(0, nullptr, &numPlatforms);
std::vector<cl_platform_id> platforms(numPlatforms);
errorCode = clGetPlatformIDs(numPlatforms, platforms, nullptr);
```
This is common pattern to get a list of things from OpenCL API.

Also it might worth moving these common test infrastructure to the header file to not duplicate this code in other tests.

Reply:
This should be:

```
errorCode = clGetPlatformIDs(numPlatforms, platforms.data(), nullptr);
```

right?